app/controllers/users/edit_email_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Users
+  class EditEmailController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def edit
+      @update_user_email_form = UpdateUserEmailForm.new(current_user)
+    end
+
+    def update
+      @update_user_email_form = UpdateUserEmailForm.new(current_user)
+
+      if @update_user_email_form.submit(user_params)
+        process_updates
+        bypass_sign_in current_user
+      else
+        render :edit
+      end
+    end
+
+    private
+
+    def user_params
+      params.require(:update_user_email_form).permit(:email)
+    end
+
+    def process_updates
+      if @update_user_email_form.email_changed?
+        track_email_change
+        flash[:notice] = t('devise.registrations.email_update_needs_confirmation')
+      end
+
+      redirect_to profile_url
+    end
+
+    def track_email_change
+      if !@update_user_email_form.email_taken?
```

Comment:
I don't understand.  There is only one condition here.  Putting the most frequently hit block first it not (IIUI) an optimization that make sense in ruby, right?  Even in C, the best thing you might do is use a hint to the branch predictor, but that would only make sense in the lowest of lowest level code.